### ISSUE TITLE
Workflow Collection event not triggering

### ISSUE THREAD

**Issue by kanodi** [nocobase/nocobase#4848]
[
![Screenshot 2024-07-08 172958](https://github.com/nocobase/nocobase/assets/72760018/dfb496a2-7845-485a-8125-3b3d95b1b92f)
](url)

Hi,

I am creating a simple demo of a quoting system, where I have quotes, which have one or more quote items. The quote items have a product, quantity, and price field.

The product collection itself also has a price field.

I am trying to create a workflow that when a quote item record is created it populates a it's price field with the products price field. To give the user the option to overwrite the price if they require.

If you see on the screenshot, they choose a product, I would like it to populate the price field from the chosen products price field on submit. But the collection event for this quote items never gets triggered.

Is this something to do with the way sub-tables work?

Thanks

**Comment by hongboji** [comment]
Yes, workflows created by sub-tables and using sub-table components to add data from sub-tables will not trigger this workflow. 
Apply for the demo on the official website, refer to the page for orders in the demo

https://github.com/nocobase/nocobase/assets/116709317/3ef5803b-0877-40bf-808a-294cd6935086

**Comment by kanodi** [comment]
thank you very much for that. Your approach above is much better/cleaner anyway. I did actually try this approach, but for some reason the 'Current Object' is the parent 'Quote' rather than the Quote Item, and no matter what I try I can't seem to change this?

**Comment by hongboji** [comment]
It's not clear how your datasheets and pages are configured to determine what issues are occurring.
Can you provide more information and a minimal reproduction process description.
